File: AMoAdiOSSdkDemo(Objective-C)/AMoAdiOSSdkDemo(Objective-C)/NativeAppViewController.m
//
//  NativeAppViewController.m
//  AMoAdiOSSdkDemo(Object-C)
//
//  Created by AMoAd on 2016/09/12.
//  Copyright © 2016年 AMoAd. All rights reserved.
//

#import "NativeAppViewController.h"
#import "AMoAdNativeView.h"

@interface NativeAppViewController () <AMoAdNativeAppDelegate>

@end

@implementation NativeAppViewController
static NSString *const kCell = @"AdViewCell";
static NSString *const kTag = @"ad01";
- (void)viewDidLoad {
    [super viewDidLoad];
    [self initAd];
}

- (void)initAd {
    UIView *templateView = [[UINib nibWithNibName:kCell bundle:nil] instantiateWithOwner:self options:nil][0];
    [templateView setFrame:CGRectMake(0, 200, 320, 100)];
    
    [[AMoAdNativeViewManager sharedManager]prepareAdWithSid:self.sid iconPreloading:true imagePreloading:true];
    [[AMoAdNativeViewManager sharedManager]renderAdWithSid:self.sid tag:kTag view:templateView delegate:self];
    
    [self.view addSubview:templateView];
}

- (void)updateAd{
    [[AMoAdNativeViewManager sharedManager] updateAdWithSid:self.sid tag:kTag];
}

- (void)clearAd {
    [[AMoAdNativeViewManager sharedManager] clearAdWithSid:self.sid tag:kTag];
}

- (IBAction)onUpdateBtnClicked:(id)sender {
    [self updateAd];
}
- (IBAction)onClearBtnClicked:(id)sender {
    [self clearAd];
}

- (void)amoadNativeDidReceive:(NSString *)sid tag:(NSString *)tag view:(UIView *)view state:(AMoAdResult)state {
    NSLog(@"amoadNativeDidReceive:%@ tag:%@ view:%@ state:%ld", sid, tag, view, (long)state);
}

- (void)amoadNativeIconDidReceive:(NSString *)sid tag:(NSString *)tag view:(UIView *)view state:(AMoAdResult)state {
    NSLog(@"amoadNativeIconDidReceive:%@ tag:%@ view:%@ state:%ld", sid, tag, view, (long)state);
}

- (void)amoadNativeImageDidReceive:(NSString *)sid tag:(NSString *)tag view:(UIView *)view state:(AMoAdResult)state {
    NSLog(@"amoadNativeImageDidReceive:%@ tag:%@ view:%@ state:%ld", sid, tag, view, (long)state);
}

- (void)amoadNativeDidClick:(NSString *)sid tag:(NSString *)tag view:(UIView *)view {
    NSLog(@"amoadNativeDidClick:%@ tag:%@ view:%@", sid, tag, view);
}
@end

File: AMoAdiOSSdkDemo/AMoAdiOSSdkDemo/NativeAppViewController.swift
//
//  NativeAppViewController.swift
//  AMoAdiOSSdkDemo
//
//  Created by AMoAd on 2016/08/16.
//  Copyright © 2016年 AMoAd. All rights reserved.
//

import UIKit

class NativeAppViewController: UIViewController, AMoAdNativeAppDelegate {
    var sid:String!
    let tag = "ad01"
    override func viewDidLoad() {
        super.viewDidLoad()
        initAd()
    }

    func initAd(){
        let templateView = UINib(nibName: "AdViewCell", bundle: nil).instantiate(withOwner: self, options: nil)[0] as! UIView
        templateView.frame = CGRect(x: 0, y: 200, width: 320, height: 100)

        AMoAdNativeViewManager.shared().prepareAd(withSid: sid, iconPreloading: true, imagePreloading: true)
        AMoAdNativeViewManager.shared().renderAd(withSid: sid, tag: tag, view: templateView, delegate: self)
        
        self.view.addSubview(templateView)
    }
    
    func updateAd() {
        AMoAdNativeViewManager.shared().updateAd(withSid: sid, tag: tag)
    }
    
    func clearAd() {
        AMoAdNativeViewManager.shared().clearAd(withSid: sid, tag: tag)
    }
    @IBAction func onUpdateBtnClicked(_ sender: AnyObject) {
        updateAd()
    }
    @IBAction func onClearBtnClicked(_ sender: AnyObject) {
        clearAd()
    }
    
    func amoadNativeDidReceive(_ sid: String!, tag: String!, view: UIView!, state: AMoAdResult) {
        print("amoadNativeDidReceive(\(sid), \(tag), \(state))")
    }
    func amoadNativeIconDidReceive(_ sid: String!, tag: String!, view: UIView!, state: AMoAdResult) {
        print("amoadNativeIconDidReceive(\(sid), \(tag), \(state))")
    }
    func amoadNativeImageDidReceive(_ sid: String!, tag: String!, view: UIView!, state: AMoAdResult) {
        print("amoadNativeImageDidReceive(\(sid), \(tag), \(state))")
    }
    func amoadNativeDidClick(_ sid: String!, tag: String!, view: UIView!) {
        print("amoadNativeDidClick(\(sid), \(tag))")
    }
}
